src/basic/fileio.c

```diff
@@ -167,6 +167,10 @@ int write_string_file_ts(
                 }
         }
 
+        if (flags & WRITE_STRING_FILE_DISABLE_BUFFER) {
+            setvbuf(f, NULL, _IONBF, 0);
+        }
```

Comment:
please follow coding style: multiples of 8 spaces indenting, and no {} around single-line if blocks. Also see CODING_STYLE.